cmd/cstorvolumeclaim/controller.go

```diff
@@ -385,3 +376,192 @@ func BaseLabels(cvc *apis.CStorVolumeClaim) map[string]string {
 	}
 	return base
 }
+
+// cvcNeedResize returns true if a cvc desired a resize operation.
+func (c *CVCController) cvcNeedResize(cvc *apis.CStorVolumeClaim) bool {
+
+	desiredCVCSize := cvc.Spec.Capacity[corev1.ResourceStorage]
+	actualCVCSize := cvc.Status.Capacity[corev1.ResourceStorage]
+
+	return desiredCVCSize.Cmp(actualCVCSize) > 0
+}
+
+// resizeCVC will:
+// 1. Mark cvc as resizing.
+// 2. Resize the cstorvolume object.
+// 3. Mark cvc as resizing finished
+func (c *CVCController) resizeCVC(cvc *apis.CStorVolumeClaim) error {
+	var updatedCVC *apis.CStorVolumeClaim
+	var err error
+	cv, err := c.clientset.OpenebsV1alpha1().CStorVolumes(cvc.Namespace).
+		Get(cvc.Name, metav1.GetOptions{})
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("falied to get cv %s: %v", cvc.Name, err))
+		return err
+	}
+	desiredCVCSize := cvc.Spec.Capacity[corev1.ResourceStorage]
+
+	if (cv.Spec.Capacity).Cmp(cv.Status.Capacity) != 0 &&
+		desiredCVCSize.Cmp(cv.Spec.Capacity) != 0 {
```

Comment:
I think here if cvc spec is updated with lower size then we are triggering resize `desiredCVCSize.Cmp(cv.Spec.Capacity) > 0`